core/vm/interpreter.go

```diff
@@ -113,6 +114,10 @@ func (in *Interpreter) Run(snapshot int, contract *Contract, input []byte) (ret
 	in.evm.depth++
 	defer func() { in.evm.depth-- }()
 
+	// Reset the previous call's return data. It's unimportant to preserve the old buffer
+	// as every returning call will return new data anyway.
+	in.returnData = nil
```

Comment:
Minor nitpick, it's not only "unimportant to preserve the old buffer", it's _required_ to clear the buffer here - otherwise a child-call could access the returndata from a parent context.